hotel-reservation-system/src/components/Authorization/FastRegister.js

```diff
@@ -0,0 +1,176 @@
+import Button from "@material-ui/core/Button";
+import CssBaseline from "@material-ui/core/CssBaseline";
+import TextField from "@material-ui/core/TextField";
+import Grid from "@material-ui/core/Grid";
+import Box from "@material-ui/core/Box";
+import Typography from "@material-ui/core/Typography";
+import { makeStyles } from "@material-ui/core/styles";
+import Container from "@material-ui/core/Container";
+import { Formik, Form, ErrorMessage, Field } from "formik";
+import {  useSelector } from "react-redux";
+import api from "./../../api/";
+import { FAST_REGISTER_VALIDATIOM_SCHEMA } from "../../constants/ValidationSchemas";
+import CallAlert from "../../Notifications/NotificationHandler";
+import { EMAIL_REGEX } from "../../constants/Regex";
+import { FillStorage, FillLocalStorage } from "./TokenData";
+
+const useStyles = makeStyles((theme) => ({
+  paper: {
+    marginTop: theme.spacing(6),
+    display: "flex",
+    flexDirection: "column",
+    alignItems: "center",
+  },
+  avatar: {
+    margin: theme.spacing(1),
+    backgroundColor: theme.palette.secondary.main,
+  },
+  form: {
+    width: "100%",
+    marginTop: theme.spacing(3),
+  },
+  submit: {
+    margin: theme.spacing(3, 0, 2),
+  },
+}));
+
+export default function FastRegister({ handleClose }) {
+  const classes = useStyles();
+  const [emailErrorLabel, setEmailErrorLabel] = useState("");
+  const [email, setEmail] = useState("");
+  const role = useSelector((state) => state.tokenData.role);
+
+  const initialValues = {
+    email: "",
+    password: "",
+    passwordConfirm: "",
+  };
+  const onSubmit = (values) => {
+    const request = {
+      Email: email,
+      Password: values.password,
+    };
+    api
+      .post("/account/register", request)
+      .then((response) => response.data)
+      .then((data) => {
+        FillStorage(data[0]);
+        FillLocalStorage(data[0], data[1]);
+        handleClose();
+        CallAlert(
+          true,
+          "you successfully registered, now you can order this room"
+        );
+      })
+      .catch((error) => {
+        if (!!error.response) {
+          setEmailErrorLabel(error.response.data.Message);
+        }
+      });
+  };
+  function ValidateEmail(email) {
+    setEmail(email);
+    setEmailErrorLabel("");
+    if (email === "") {
+      setEmailErrorLabel("email is reqired");
+      return false;
+    }
+    const flag = EMAIL_REGEX.test(email);
+    if (!flag) {
+      setEmailErrorLabel("invalid email");
+      return false;
+    }
+    setEmail(email);
+    return true;
+  }
+
+  return (
+    <Container component="main" maxWidth="xs">
+      <CssBaseline />
+      <div className={classes.paper}>
+        <Typography component="h1" variant="h5">
+          Fast Register
+        </Typography>
+        <Formik
+          initialValues={initialValues}
+          onSubmit={(values) => {
+            if (ValidateEmail(email.trim()) && emailErrorLabel === "")
+              onSubmit(values);
```

Comment:
Please move this to the function handleSubmit

Reply:
moved